lib/galaxy/webapps/galaxy/controllers/dataset.py

```diff
@@ -975,7 +975,12 @@ def _purge(self, trans, dataset_id):
             # HDA is purgeable
             # Decrease disk usage first
             if user:
-                user.adjust_total_disk_usage(-hda.quota_amount(user))
+                if len(hda.dataset.active_storage_media_associations) == 0:
```

Comment:
See comment about API controller layer above. This shouldn't be happening in this controller I don't think.

Reply:
Updated to use the property instead.